### PR TITLE
Add documentation for docker integration

### PR DESCRIPTION
**Pull request by factyy**
Both for `lsp-mode` and `dap-mode`. 
As was requested by @yyoncho in emacs-lsp/dap-mode#559 

Hopefully the `dap-mode` PR (linked above) will be merged too

Sorry for the delay!

### CODE REVIEW

File: docs/tutorials/docker-integration.md [hunk 1]
```diff
@@ -0,0 +1,161 @@
+# Docker integration
```
Comment (ericdallo) [inline docs/tutorials/docker-integration.md:1]: WDYT about adding a header with you as @author just like other tutorials?

Comment (factyy) [reply docs/tutorials/docker-integration.md:1]: Added in [8783e99](https://github.com/emacs-lsp/lsp-mode/pull/3448/commits/8783e99dadb8d0a9212b988f540f88f1f471da18)

**Comment by yyoncho** [comment]
Thank you for the docs. It will really help with testing the PR. I will try to do that during the weekend.

**Comment by yyoncho** [comment]
@factyy Sorry, I am very busy lately (new job, newborn, etc). I haven't forgotten about this PR, but these days I don't have much time/energy. I will try to do the testing by the end of the week.

**Comment by yyoncho** [comment]
@factyy thank you, and sorry for the delay.

**Pull request merged**